os/kernel/log_dump/log_dump.c

```diff
@@ -239,3 +439,46 @@ size_t log_dump_read(FAR char *buffer, size_t buflen)
 
 	return ret;
 }
+
+int log_dump(int argc, char *argv[])
+{
+	struct mq_attr attr;
+	mqd_t mq_fd;
+
+	attr.mq_maxmsg = 32;
+	attr.mq_msgsize = sizeof(struct log_dump_msg_s);
+	attr.mq_flags = 0;
+
+	/* Create log dump message queue */
+	mq_fd = mq_open(LOG_DUMP_MSGQ, O_RDWR | O_CREAT, 0666, &attr);
+	if (mq_fd == (mqd_t) ERROR) {
+		lldbg("Fail to open mq, errno %d. EXIT!\n", errno);
+		return 0;
+	}
+
+	if (log_dump_init() != LOG_DUMP_OK) {
+		lldbg("Fail to init log dump\n");
+		mq_close(mq_fd);
+		return 0;
+	}
+
+	struct log_dump_msg_s msg;
+	int nbytes;
+
+	while (true) {
+		nbytes = mq_receive(mq_fd, (char *)&msg, sizeof(struct log_dump_msg_s), NULL);
+		if (nbytes <= 0) {
+			continue;
+		}
+		if (msg.comp) {
+			log_dump_read_init();
+			last_block_compression = false;
+		} else {
+			compress_ret = compress_block(out_buf, &writesize, uncomp_array, CONFIG_LOG_DUMP_CHUNK_SIZE);
+			if (compress_ret != LOG_DUMP_OK) {
+				lldbg("Fail to compress compress_ret = %d\n", compress_ret);
+			}
+			current_compression = false;
+		}
```

Comment:
Could you explain if and else case?
The `log_dump_read_init` calls `compress_block`, but else also calls it.

Reply:
if used to compress last partially filled block and else for compression entire 4k filled blocks